lms/djangoapps/courseware/user_state_client.py

```diff
@@ -253,7 +253,7 @@ def delete_many(self, username, block_keys, scope=Scope.user_state, fields=None)
         student_modules = self._get_student_modules(username, block_keys)
         for student_module, _ in student_modules:
             if fields is None:
-                student_module.state = None
+                student_module.state = "{}"
```

Comment:
So an empty student_module.state is always represented in the DB with a `{}`? It should never be `NULL`?